pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -117,3 +126,86 @@ func (fc *fluxForCluster) writeFluxUpgradeFiles() error {
 
 	return nil
 }
+
+func (fc *fluxForCluster) updateFileStructure() error {
+	config := fc.clusterSpec.GitOpsConfig
+	clusterRoot := config.Spec.Flux.Github.ClusterRootPath()
+	fluxSystemPath := filepath.Join(fc.gitOpts.Writer.Dir(), fc.fluxSystemDir())
+	nEksaSystemPath := filepath.Join(fc.gitOpts.Writer.Dir(), fc.eksaSystemDir())
+	oEksaSystemPath := filepath.Join(fc.gitOpts.Writer.Dir(), clusterRoot, eksaSystemDirName)
+
+	if validations.FileExists(fluxSystemPath) && validations.FileExists(nEksaSystemPath) {
+		logger.V(3).Info("File structure is already up-to-date")
+		return nil
+	}
+
+	if !validations.FileExists(nEksaSystemPath) && validations.FileExists(oEksaSystemPath) {
+		logger.V(3).Info("Updating eksa-system files...")
+		if err := fc.writeEksaUpgradeFiles(oEksaSystemPath); err != nil {
+			return err
+		}
+		err := fc.gitOpts.Git.Remove(filepath.Join(clusterRoot, eksaSystemDirName))
+		if err != nil {
+			return &ConfigVersionControlFailedError{Err: fmt.Errorf("error when removing %s in git: %v", filepath.Join(clusterRoot, eksaSystemDirName), err)}
+		}
+	}
+
+	return nil
+}
+
+func (fc *fluxForCluster) writeEksaUpgradeFiles(oldEksaSystemPath string) error {
+	fileName := filepath.Join(oldEksaSystemPath, clusterConfigFileName)
+	resourcesSpec, err := fc.updateGitOpsConfig(fileName)
+	if err != nil {
+		return err
+	}
+	w, err := fc.initEksaWriter()
+	if err != nil {
+		return err
+	}
+
+	t := templater.New(w)
+	logger.V(3).Info("Updating eksa-system cluster config file...")
+	if _, err = t.WriteBytesToFile(resourcesSpec, clusterConfigFileName, filewriter.PersistentFile); err != nil {
+		return err
+	}
+
+	return fc.generateEksaKustomizeFile()
```

Comment:
This will generate the kustomize files with the new spec
Can't we just move the original files?

Reply:
don't think so -- the kustomization of eks-a shall follows the file structure changes